---
title: "MICB425 Portfolio - Microbial Ecological Genomics: From Genomes to Biomes, Spring 2018"
author: "Jinsong, Tony, Yan, 18752148"
date: "Version 1.0 (January 19, 2018)"
output: html_document
---

# Module 01
Reserve the first level headings (#) for the start of a new Module. This will help to organize your portfolio in an intuitive fashion.  

## Module 01 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.

* Installation check
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty html Challenge
    + Completion status:
    + Comments:
* Evidence worksheet_01
    + Completion status:
    + Comments:
* Evidence worksheet_02
    + Completion status:
    + Comments:
* Evidence worksheet_03
    + Completion status:
    + Comments:
* Problem Set_01
    + Completion status:
    + Comments:
* Problem Set_02
    + Completion status:
    + Comments:
* Writing assessment_01
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments  
    
## Data science Friday  
The remaining second level headers (##) are for separating data science Friday, regular course, and project content. In this module, you will only need to include data science Friday and regular course content; projects will come later in the course.


### Installation check
Use this space to include your installation screenshots.

![Terminal](https://github.com/tonyyan996/MICB425_portfolio/blob/master/Screenshot1.png?raw=true)

![Rstudio](https://github.com/tonyyan996/MICB425_portfolio/blob/master/Screenshot2.png?raw=true)

![Github](https://github.com/tonyyan996/MICB425_portfolio/blob/master/Screenshot3.png?raw=true)

### Portfolio repo setup
Detail the code you used to create, initialize, and push your portfolio repo to GitHub. This will be helpful as you will need to repeat many of these steps to update your porfolio throughout the course.

stndin: git add .

stndin: git commit -m "commit message here"

stndin: git status

stndin: git push

--------------

## Origins and Earth Systems
### Evidence Worksheet for "Prokaryotes: The Unseen Majority"

### Learning Objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems.

### General Questions
#### What were the main questions being asked?

Answer: How many prokaryotes are there in the world? How abundant are they relative to their ecosystems? To what extent do they participate in the global carbon cycling (and nitrogen, phosphorus, etc)?

#### What were the primary methodological approaches used?

Answer: 
Estimation Methods:
Water: Sediment density/mL
Soil: Estimated from unpublished field studies. 
Subsurface: Ground water sample estimations,as well as estimations from porosity of the ground and amount of space that prokaryotes occupied in pores. 
Microbiota: Estimation from microbiota density and the population sizes of animals. 
Leaves/Plants: Estimates are made from leaf area estimated by leaf area index. 
Carbon estimations: Total organisms multiplied by 1/2 dry weight. 

#### Summarize the main results or findings. 

Answer:
Essentially nutrient cycle estimations for a variety of environments.
Total cellular carbon: 350-550 Pg of C
Total cells: 4-6 x 10^30
Total cellular nitrogen: 85-130 Pg N
Total cellular phosphorus: 9-14 Pg P
Total prokaryotic microorganisms:
Open Ocean: 1.2 x 10^29
Soil: 2.6 x 10^29
ocean/terrestrial subsurface: 3.5 x 10^30	0.25-2.5 x 10^30
Report statistics about how many prokaryotes are located in different environments. 

#### Do new questions arise from the results?

Answer: How does the abundance of the prokaryotes in these environments play a role in the total metabolic potential in their ecosystem? How did the authors compile their calculations into their finalized findings, why did they use the assumptions that they did in doing so? How can the authors be sure that their estimations are accurate and precise? Since they have no discrete controls or replicates? What were the statistical methods used to standardize across environments, between different sampling methodologies?

#### Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

Methods were not fully explained to a point of possible replication of their estimates. 
Authors did not provide enough justification to use the assumptions that they did, and does not show how their assumptions may affect their estimates in one way or another.

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)
---------------

### Problem set 01
#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:
* What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  

  __(Answer): Marine environments (1.181x10^29) including polar water-ice interfaces, Soil (2.556x10^29), subsurface marine sediments (3.8x10^30), and microbiota.__

* What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  

  __(Answer): Upper 200m of ocean: Total number of cells = 3.6x10^28. Versus 2.9x10^27 for autotrophic prochlorococcus. Autotrophs have approximately 8% (4x10^4 / 5x10^5) of the total cell density for upper 200m ocean, meaning that carbon fixation makes up 80% of carbon cycling abundance in ocean, indicating a significant amount of the Earth's carbon dioxide is fixed through the ocean.__

* What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?  

  __(Answer): Autotroph: Fixes carbon from carbon dioxide gas. Heterotroph: Uses carbon from organic compounds. Lithotroph: Uses inorganic substances (like minerals) to obtain reducing agents.__

* Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  

  __(Answer): Deep habitats supporting life, subsurface separated into terrestrial and marine. Temperature is limiting factor at very high depths in Mariana Trench (10.9km).__

* Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?  

  __(Answer): Qualities of air and atmosphere at that environment (Mt Everest 8.8km) also affects growth capabilities.__

* Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth’s biosphere measured in km?  

  __(Answer): 24km from the highest point to the lowester point on Earth represents the general biosphere.__

* How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)  

  __(Answer): Population x turnover/yr = cells/yr. e.g. marine heteros -> [3.6x10^28 cells x 365 days]/16 turnovers = 8.2x10^29 cells/yr. Virus affect growth and turnover rate and may contain metabolic assessory genes that can reprogram cells metabolically.__

* What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  

  __(Answer): Carbon Efficiency = 20%  5-20fg C/cell 10x10^-30 pg/Cell  (3.6x10^28 Cells)x(10x10^-30 pg/cell)= 0.72 Pg carbonin marine heteros. 4x0.72= 2.88 Pg/yr of C. 51 Pg C/yr w/ 85% consumed = 43 Pg C/yr (43 Pg C/yr)/(2.88 Pg/yr) = 14.9, 1 turnover every 24.5 days.__

* How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  

  __(Answer): (4x10^-7 mutations/generation)^4 (simultaneous mutations) = 2.56x10^26 mutations/generation. 3.6x10^28 cells and 365/16=22.8 turnovers/yr. (3.6x10^28 x 22.8 = 8.2x10^29 Cells/yr x 2.56x10^-26 mutations/generation) = 2.1x10^4 mutations/yr.__

* Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt? 

  __(Answer): Microbial genomes are horizontally trasferrable, indicating that a higher population size and population diversity will make the community as a whole, more dynamic and flexible to change. When under selection pressure due to a sudden drastic environment shift, microbial genetic diversity and HGT mechanisms ensure that genes necessary for fitness (selected from the few species that may possess them) is passed around the community quickly. __

* What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text? 

  __(Answer): Prokaryotic abundance corresponds more towards the magnitude of metabolic activity in an environment, characterizing the total capacity for the ecosystem to undergo biochemical reactions. Prokaryotic species diversity is logarithmically related to metabolic diversity of the environment, where a greater species diversity contributes to a more diverse reaction sets that could potentially take place, up to an extent where eventually greater species diversity will result in metabolic redundancy.__

---------

#Pretty_html
###_Tony Yan (18752148)_
###_version 1/24/2018_

  * [R Markdown PDF Challenge](#r markdown pdf challenge)
    + [Challenge Goals](##challenge goals)
        * [Here's a header!](###heres a header)
        * [Table Time](###Table Time)

#R Markdown PDF Challenge
The following assignment is an exercise for the reproduction of this .html document using the RStudio and RMarkdown tools we’ve shown you in class. Hopefully by the end of this, you won’t feel at all the way this poor PhD student does. We’re here to help, and when it comes to R, the internet is a really valuable resource. This open-source program has all kinds of tutorials online.
![](https://pbs.twimg.com/media/B_NkGp5VEAA_YVh.jpg)

[http://phdcomics.com/](http://phdcomics.com/) Comic posted 1-17-2018

##Challenge Goals
The goal of this R Markdown html challenge is to give you an opportunity to play with a bunch of different RMarkdown formatting. Consider it a chance to flex your RMarkdown muscles. Your goal is to write your own RMarkdown that rebuilds this html document as close to the original as possible. So, yes, this means you get to copy my irreverant tone exactly in your own Markdowns. It’s a little window into my psyche. Enjoy =)

__hint: go to the [PhD Comics website](http://phdcomics.com/) to see if you can find the image above__

_If you can’t find that exact image, just find a comparable image from the PhD Comics website and include it in your markdown_

### Here's a header!
Let’s be honest, this header is a little arbitrary. But show me that you can reproduce headers with different levels please. This is a level 3 header, for your reference (you can most easily tell this from the table of contents).

####Another Header, now with maths
Perhaps you’re already really confused by the whole markdown thing. Maybe you’re so confused that you’ve forgotton how to add. Never fear! ~~A calculator~~ R is here:
```{r eval=FALSE}
1231521+12341556280987
```
```{r echo=FALSE}
1231521+12341556280987
```

###Table Time
Or maybe, after you’ve added those numbers, you feel like it’s about time for a table!
I’m going to leave all the guts of the coding here so you can see how libraries (R packages) are loaded into R (more on that later). It’s not terribly pretty, but it hints at how R works and how you will use it in the future. The summary function used below is a nice data exploration function that you may use in the ^future^.

```{r}
library(knitr)
kable(summary(cars),caption="I made this table with kable in the knitr package library")
```

And now you’ve almost finished your first RMarkdown! Feeling excited? We are! In fact, we’re so excited that maybe we need a big finale eh? Here’s ours! Include a fun gif of your choice!

![](https://media.giphy.com/media/vFKqnCdLPNOKc/giphy.gif)

----------

### Evidence Worksheet_02 “Life and the Evolution of Earth’s Atmosphere”
#### Learning objectives: 
Comment on the emergence of microbial life and the evolution of Earth systems
    
    
* Indicate the key events in the evolution of Earth systems at each approximate moment in the time series. If times need to be adjusted or added to the timeline to fully account for the development of Earth systems, please do so. 

    + 4.6 billion years ago: Formation of Earth and the moon from the proposed collision between proto-Earth and proto-moon. Caused formation of Earth's early atmosphere, comprised of mostly methane, ammonia, hydrogen, nitrogen, and water vapor. 
    
    + 4.2 billion years ago: Earliest evidence for life. Late heavy bombardment.
    
    + 3.8 billion years ago: Earliest signs of photosynthesis. Rubisco discovered. Early formation of oceans.
    
    + 3.5 billion years ago: Earliest common ancestor between bacteria and archea split. Earliest fossils of bacteria and archaea dates to here. Anoxic photogenesis from microfossils. 
  
    + 3.0 billion years ago: First large asteroid impact on Earth. Photosynthetic cyanobacteria evolve. Increase in oxygenation of the atomosphere. First global glaciation. 
    
    + 2.7 billion years ago: Early evidence of eukaryotes discovered by association with sterols. Saturation of ocean with oxygen, significant oxygen presence in atmosphere. Early life on land. 
    
    + 2.2 billion years ago: Earliest known eukaryotic fossils.   
    
    + 2.1 billion years ago: Lesser early supercontinents form
    
    + 1.8 billion years ago: Stablization of second continental mass. Early evidence of oxidation corresponding to rise in oxygen levels in atmosphere. Complex eukaryotes evolved. 
    
    + 1.3 billion years ago: Earliest evidence of sexually reproducing organisms from single cell eukarya. Early evidence for predation. 
    
    + 1.0 billion years ago: First major ice age, glaciation disrupts carbon cycle. 
    
    + 550 million years ago: First forms of skeletal life. End of "snowball Earth", earliest arthropods and fungi. Cambrian explosion, sees first emergence of many vertebrates. 
    
    + 350 million years ago: Evolution diversifies species of vertebrates. Further increase in oxygenation of atomosphere due to emergence of woody land plants. Pangea emerges. 
    
    + 200 million years ago: Dinosaurs! Followed by Permian extinction, loss of 95% species diversity. 
    
    + 60 million years ago: Evolution of mammals and other vertebrates. Dramatic global warming followed by ice age.
    
    + 2 million years ago - today: Evolution of humans and mammals post ice age. 


* Describe the dominant physical and chemical characteristics of Earth systems at the following waypoints:  

    + Hadea: Ball of molten atmosphere and molten crust.
    
    + Archean: Early oxgenation of the atmosphere with photosynthesis prokaryotes. High concentration of hydrocarbons in atmosphere. Sun was 30% less luminous. 
    
    + Precambrian: End of snowball Earth. Species see explosion in diversity. Atmosphere and geological properties support different metabolic processes and nutrient cycling at the most basic level. 
    
    + Proterozoic: Plants and animals differentiate, causing environmental conditions to support autotrophs and heterotrophs. Greater oxygenation of the atmosphere to support multicellular respiration. 
    
    + Phanerozoic: Global warming and cooling cycles between ice age and warming. Causing bloom in species diversity followed by mass extinctions. Atmosphere and geologic conditions become rich in organic nutrients quickly giving birth to new species when the niche is right. Emergence of complex carbon, oxygen, nitrogen, phosphorous, and sulfur cycling. 
    
-----------------

### Problem set 02
#### Learning objectives: 
Discuss	 the	 role	of	microbial	diversity and	 formation	of	coupled	metabolism	in	driving
global	biogeochemical	cycles.

#### General questions:
* What	 are	 the	 primary	 geophysical	 and	 biogeochemical	 processes	 that	 create	 and	 sustain	conditions	for	life	on	Earth? How	do	abiotic	versus	biotic	processes vary	with	respect	to	matter and	energy	transformation and	how	are	they	interconnected?

  __(Answer): Volcanism and rock weathering and plate tectonics helped to shape the geological conditions for life. Supplies and sinks carbon dioxide and sulfur cycling. __

* Why	is Earth’s	redox	state	considered	an emergent	property?

  __(Answer): Life on earth cannot be attributed to a single factor, therefore, the entire system ought to be examined as a whole. Where the whole is more than a sum of its parts. Many organisms needs to be coupled to other organisms where one biological process can only occur when linked to another.__
  
* How	do reversible	electron transfer	reactions	give rise	to	element and nutrient cycles at different	ecological scales? What	strategies do	microbes use	to	overcome thermodynamic barriers to	reversible electron	flow?

__(Answer): Thermodynamics determines favorable flow of electrons that allows biological reactions to occur spontaneously. Each reaction is spontaneous only in the environment which it occurs in, and in order to revert the reaction, the environmental conditions need to change for the reverse reaction to be spontaneous. This causes nutrients to be generally localized, preventing a global even distribution of nutrients in all environments.__

* Using information provided in the text, describe how the nitrogen cycle	partitions between	different redox “niches” and microbial groups. Is there a relationship between the nitrogen	cycle	and climate	change?	

__(Answer): Microbes generally form distirbuted groups that carry out certain metabolic functions within their communities. These groups specialized into niches that carry out only one reaction within the nitrogen cycle. For example a group of organisms that reduces N2 gas to ammonium, and the reverse reaction of oxidizing ammonium are carried out by niches of other communities. Nitrous oxide is a potent greenhouse gas.__

* What is	the	relationship between	microbial	diversity	and	metabolic	diversity	and	how	does this relate	to	the	discovery	of new	protein	families from	microbial	community	genomes?

__(Answer): Much of metabolic diversity arises from the adaptation of existing metabolic pathways due to great correlation and co-evolution between microbial species. Microorganisms utilize available resources from their environment in the most thermaldynamically favorable means, which often times result in modulation and altered implementation of existing pathways and enzymes, rather than evolving entirely novel mechanisms. This can be done over time through horizontal gene transfers between species, and emergence of new species from existing ones based on genetic differentiation. As species diversity increases, metabolic diversity increases logarithmically, meaning fewer and fewer new proteins without homologs are discovered for every species discovered.__

* On	what	basis	do	the	authors	consider	microbes	the	guardians	of	metabolism?

__(Answer): The Authors claim that due to horizontal gene transfer and the general conserved nature of genes that encode metabolic pathway enzymes, microbes that may survive a certain selection pressure can go on to preserve their metabolic capabilities and transfer them to the overall environment. Regardless of what the selection pressures are or how strong they are, certain species will always survive, and they serve as a sort of noah's ark for metabolic genes so that the environment may continue to cycle nutrients.__

-----------------

# Working with Data in R (Data science Friday 4)
###_Tony Yan (18752148)_
###_version 1/31/2018_

### Exercise 1:
_We have another data table we will be using in up-coming DS Fridays, Saanich.OTU.txt. Import this data as a named object with the correct parameters in R._
```{r tables}
read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings=c("NAN", "NA", "."))
Saanich_metadata = read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings=c("NAN", "NA", "."))
```
### Exercise 2:
_Using dplyr, find at what depth(s) methane (CH4) is above 100 nM while temperature is below 10 °C. Print out a table showing only the depth, methane, and temperature data._
```{r dplyr}

library(dplyr)

select(Saanich_metadata, starts_with("CH4"))
filter(Saanich_metadata, CH4_nM > 100 & Temperature_C < 10) %>% select(Depth_m)
```
### Exercise 3:
_Convert all variables that are in nM to μM. Output a table showing only the original nM and converted μM variables._
```{r dplyr_tables}

library(dplyr)

select(Saanich_metadata, ends_with("nM")) %>% 
  mutate(N2O_uM = N2O_nM/1000) %>% mutate(Std_N2O_uM = Std_N2O_nM/1000) %>% mutate(CH4_uM = CH4_nM/1000) %>% mutate(Std_CH4_uM = Std_CH4_nM/1000) 

```

----------------

### Module 01 references
Utilize this space to include a bibliography of any literature you want associated with this module. We recommend keeping this as the final header under each module.

Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578–6583.  [PMC33863](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

