---
title: "MICB425 Portfolio - Microbial Ecological Genomics: From Genomes to Biomes, Spring 2018"
author: "Jinsong, Tony, Yan, 18752148"
date: "Version 1.0 (January 19, 2018)"
output: html_document
---

# Module 01
Reserve the first level headings (#) for the start of a new Module. This will help to organize your portfolio in an intuitive fashion.  

## Module 01 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.

* Installation check
    + Completion status:
    + Comments:
* Portfolio repo setup
    + Completion status: 
    + Comments:
* RMarkdown Pretty html Challenge
    + Completion status:
    + Comments:
* Evidence worksheet_01
    + Completion status:
    + Comments:
* Evidence worksheet_02
    + Completion status:
    + Comments:
* Evidence worksheet_03
    + Completion status:
    + Comments:
* Problem Set_01
    + Completion status:
    + Comments:
* Problem Set_02
    + Completion status:
    + Comments:
* Writing assessment_01
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments  
    
## Data science Friday  
The remaining second level headers (##) are for separating data science Friday, regular course, and project content. In this module, you will only need to include data science Friday and regular course content; projects will come later in the course.


## Installation check (Data Science Friday 1)
Use this space to include your installation screenshots.

![Terminal](https://github.com/tonyyan996/MICB425_portfolio/blob/master/Screenshot1.png?raw=true)

![Rstudio](https://github.com/tonyyan996/MICB425_portfolio/blob/master/Screenshot2.png?raw=true)

![Github](https://github.com/tonyyan996/MICB425_portfolio/blob/master/Screenshot3.png?raw=true)

### Portfolio repo setup (Data Science Friday 2)
Detail the code you used to create, initialize, and push your portfolio repo to GitHub. This will be helpful as you will need to repeat many of these steps to update your porfolio throughout the course.

__To intialize github account from github:__

stndin: git config --global user.name "Your Name"

stndin: git config --global user.email "youremail@email.com"

__To clone from github master:__

stndin: git clone [git url]

__To pull down from github master:__

stndin: git fetch

stndin: git pull

__To push to github from origin master:__

stndin: git add .

stndin: git commit -m "commit message here"

stndin: git status

stndin: git push

--------------

## Origins and Earth Systems
### Evidence Worksheet for "Prokaryotes: The Unseen Majority"

### Learning Objectives
Describe the numerical abundance of microbial life in relation to ecology and biogeochemistry of Earth systems.

### General Questions
#### What were the main questions being asked?

(Answer): How many prokaryotes are there in the world? How abundant are they relative to their ecosystems? To what extent do they participate in the global carbon cycling (and nitrogen, phosphorus, etc)?__

#### What were the primary methodological approaches used?

(Answer): 
Estimation Methods:
Water: Sediment density/mL
Soil: Estimated from unpublished field studies. 
Subsurface: Ground water sample estimations,as well as estimations from porosity of the ground and amount of space that prokaryotes occupied in pores. 
Microbiota: Estimation from microbiota density and the population sizes of animals. 
Leaves/Plants: Estimates are made from leaf area estimated by leaf area index. 
Carbon estimations: Total organisms multiplied by 1/2 dry weight. 

#### Summarize the main results or findings. 

(Answer):
Essentially nutrient cycle estimations for a variety of environments.
Total cellular carbon: 350-550 Pg of C
Total cells: 4-6 x 10^30
Total cellular nitrogen: 85-130 Pg N
Total cellular phosphorus: 9-14 Pg P
Total prokaryotic microorganisms:
Open Ocean: 1.2 x 10^29
Soil: 2.6 x 10^29
ocean/terrestrial subsurface: 3.5 x 10^30	0.25-2.5 x 10^30
Report statistics about how many prokaryotes are located in different environments. 

#### Do new questions arise from the results?

(Answer): How does the abundance of the prokaryotes in these environments play a role in the total metabolic potential in their ecosystem? How did the authors compile their calculations into their finalized findings, why did they use the assumptions that they did in doing so? How can the authors be sure that their estimations are accurate and precise? Since they have no discrete controls or replicates? What were the statistical methods used to standardize across environments, between different sampling methodologies?

#### Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

(Answer): Methods were not fully explained to a point of possible replication of their estimates. 
Authors did not provide enough justification to use the assumptions that they did, and does not show how their assumptions may affect their estimates in one way or another.

[Whitman *et al* 1998](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)
---------------

## Problem set 01
#### Learning objectives: 
Describe the numerical abundance of microbial life in relation to the ecology and biogeochemistry of Earth systems.

#### Specific questions:
* What are the primary prokaryotic habitats on Earth and how do they vary with respect to their capacity to support life? Provide a breakdown of total cell abundance for each primary habitat from the tables provided in the text.  

(Answer): Marine environments (1.181x10^29) including polar water-ice interfaces, Soil (2.556x10^29), subsurface marine sediments (3.8x10^30), and microbiota.

* What is the estimated prokaryotic cell abundance in the upper 200 m of the ocean and what fraction of this biomass is represented by marine cyanobacterium including Prochlorococcus? What is the significance of this ratio with respect to carbon cycling in the ocean and the atmospheric composition of the Earth?  

(Answer): Upper 200m of ocean: Total number of cells = 3.6x10^28. Versus 2.9x10^27 for autotrophic prochlorococcus. Autotrophs have approximately 8% (4x10^4 / 5x10^5) of the total cell density for upper 200m ocean, meaning that carbon fixation makes up 80% of carbon cycling abundance in ocean, indicating a significant amount of the Earth's carbon dioxide is fixed through the ocean.

* What is the difference between an autotroph, heterotroph, and a lithotroph based on information provided in the text?  

(Answer): Autotroph: Fixes carbon from carbon dioxide gas. Heterotroph: Uses carbon from organic compounds. Lithotroph: Uses inorganic substances (like minerals) to obtain reducing agents.

* Based on information provided in the text and your knowledge of geography what is the deepest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this depth?  

(Answer): Deep habitats supporting life, subsurface separated into terrestrial and marine. Temperature is limiting factor at very high depths in Mariana Trench (10.9km).

* Based on information provided in the text your knowledge of geography what is the highest habitat capable of supporting prokaryotic life? What is the primary limiting factor at this height?  

(Answer): Qualities of air and atmosphere at that environment (Mt Everest 8.8km) also affects growth capabilities.

* Based on estimates of prokaryotic habitat limitation, what is the vertical distance of the Earth’s biosphere measured in km?  

(Answer): 24km from the highest point to the lowester point on Earth represents the general biosphere.

* How was annual cellular production of prokaryotes described in Table 7 column four determined? (Provide an example of the calculation)  

(Answer): Population x turnover/yr = cells/yr. e.g. marine heteros -> [3.6x10^28 cells x 365 days]/16 turnovers = 8.2x10^29 cells/yr. Virus affect growth and turnover rate and may contain metabolic assessory genes that can reprogram cells metabolically.

* What is the relationship between carbon content, carbon assimilation efficiency and turnover rates in the upper 200m of the ocean? Why does this vary with depth in the ocean and between terrestrial and marine habitats?  

(Answer): Carbon Efficiency = 20%  5-20fg C/cell 10x10^-30 pg/Cell  (3.6x10^28 Cells)x(10x10^-30 pg/cell)= 0.72 Pg carbonin marine heteros. 4x0.72= 2.88 Pg/yr of C. 51 Pg C/yr w/ 85% consumed = 43 Pg C/yr (43 Pg C/yr)/(2.88 Pg/yr) = 14.9, 1 turnover every 24.5 days.

* How were the frequency numbers for four simultaneous mutations in shared genes determined for marine heterotrophs and marine autotrophs given an average mutation rate of 4 x 10^-7^ per DNA replication? (Provide an example of the calculation with units. Hint: cell and generation cancel out)  

(Answer): (4x10^-7 mutations/generation)^4 (simultaneous mutations) = 2.56x10^26 mutations/generation. 3.6x10^28 cells and 365/16=22.8 turnovers/yr. (3.6x10^28 x 22.8 = 8.2x10^29 Cells/yr x 2.56x10^-26 mutations/generation) = 2.1x10^4 mutations/yr.

* Given the large population size and high mutation rate of prokaryotic cells, what are the implications with respect to genetic diversity and adaptive potential? Are point mutations the only way in which microbial genomes diversify and adapt? 

(Answer): Microbial genomes are horizontally trasferrable, indicating that a higher population size and population diversity will make the community as a whole, more dynamic and flexible to change. When under selection pressure due to a sudden drastic environment shift, microbial genetic diversity and HGT mechanisms ensure that genes necessary for fitness (selected from the few species that may possess them) is passed around the community quickly.

* What relationships can be inferred between prokaryotic abundance, diversity, and metabolic potential based on the information provided in the text? 

(Answer): Prokaryotic abundance corresponds more towards the magnitude of metabolic activity in an environment, characterizing the total capacity for the ecosystem to undergo biochemical reactions. Prokaryotic species diversity is logarithmically related to metabolic diversity of the environment, where a greater species diversity contributes to a more diverse reaction sets that could potentially take place, up to an extent where eventually greater species diversity will result in metabolic redundancy.

---------

#Pretty_html (Data Science Friday 3)
###_Tony Yan (18752148)_
###_version 1/24/2018_

  * [R Markdown PDF Challenge](#r markdown pdf challenge)
    + [Challenge Goals](##challenge goals)
        * [Here's a header!](###heres a header)
        * [Table Time](###Table Time)

#R Markdown PDF Challenge
The following assignment is an exercise for the reproduction of this .html document using the RStudio and RMarkdown tools we’ve shown you in class. Hopefully by the end of this, you won’t feel at all the way this poor PhD student does. We’re here to help, and when it comes to R, the internet is a really valuable resource. This open-source program has all kinds of tutorials online.
![](https://pbs.twimg.com/media/B_NkGp5VEAA_YVh.jpg)

[http://phdcomics.com/](http://phdcomics.com/) Comic posted 1-17-2018

##Challenge Goals
The goal of this R Markdown html challenge is to give you an opportunity to play with a bunch of different RMarkdown formatting. Consider it a chance to flex your RMarkdown muscles. Your goal is to write your own RMarkdown that rebuilds this html document as close to the original as possible. So, yes, this means you get to copy my irreverant tone exactly in your own Markdowns. It’s a little window into my psyche. Enjoy =)

__hint: go to the [PhD Comics website](http://phdcomics.com/) to see if you can find the image above__

_If you can’t find that exact image, just find a comparable image from the PhD Comics website and include it in your markdown_

### Here's a header!
Let’s be honest, this header is a little arbitrary. But show me that you can reproduce headers with different levels please. This is a level 3 header, for your reference (you can most easily tell this from the table of contents).

####Another Header, now with maths
Perhaps you’re already really confused by the whole markdown thing. Maybe you’re so confused that you’ve forgotton how to add. Never fear! ~~A calculator~~ R is here:
```{r eval=FALSE}
1231521+12341556280987
```
```{r echo=FALSE}
1231521+12341556280987
```

###Table Time
Or maybe, after you’ve added those numbers, you feel like it’s about time for a table!
I’m going to leave all the guts of the coding here so you can see how libraries (R packages) are loaded into R (more on that later). It’s not terribly pretty, but it hints at how R works and how you will use it in the future. The summary function used below is a nice data exploration function that you may use in the ^future^.

```{r}
library(knitr)
kable(summary(cars),caption="I made this table with kable in the knitr package library")
```

And now you’ve almost finished your first RMarkdown! Feeling excited? We are! In fact, we’re so excited that maybe we need a big finale eh? Here’s ours! Include a fun gif of your choice!

![](https://media.giphy.com/media/vFKqnCdLPNOKc/giphy.gif)

----------

## Evidence Worksheet_02 “Life and the Evolution of Earth’s Atmosphere”
#### Learning objectives: 
Comment on the emergence of microbial life and the evolution of Earth systems
    
    
* Indicate the key events in the evolution of Earth systems at each approximate moment in the time series. If times need to be adjusted or added to the timeline to fully account for the development of Earth systems, please do so. 

    + 4.6 billion years ago: Formation of Earth and the moon from the proposed collision between proto-Earth and proto-moon. Caused formation of Earth's early atmosphere, comprised of mostly methane, ammonia, hydrogen, nitrogen, and water vapor. 
    
    + 4.2 billion years ago: Earliest evidence for life. Late heavy bombardment.
    
    + 3.8 billion years ago: Earliest signs of photosynthesis. Rubisco discovered. Early formation of oceans.
    
    + 3.5 billion years ago: Earliest common ancestor between bacteria and archea split. Earliest fossils of bacteria and archaea dates to here. Anoxic photogenesis from microfossils. 
  
    + 3.0 billion years ago: First large asteroid impact on Earth. Photosynthetic cyanobacteria evolve. Increase in oxygenation of the atomosphere. First global glaciation. 
    
    + 2.7 billion years ago: Early evidence of eukaryotes discovered by association with sterols. Saturation of ocean with oxygen, significant oxygen presence in atmosphere. Early life on land. 
    
    + 2.2 billion years ago: Earliest known eukaryotic fossils.   
    
    + 2.1 billion years ago: Lesser early supercontinents form
    
    + 1.8 billion years ago: Stablization of second continental mass. Early evidence of oxidation corresponding to rise in oxygen levels in atmosphere. Complex eukaryotes evolved. 
    
    + 1.3 billion years ago: Earliest evidence of sexually reproducing organisms from single cell eukarya. Early evidence for predation. 
    
    + 1.0 billion years ago: First major ice age, glaciation disrupts carbon cycle. 
    
    + 550 million years ago: First forms of skeletal life. End of "snowball Earth", earliest arthropods and fungi. Cambrian explosion, sees first emergence of many vertebrates. 
    
    + 350 million years ago: Evolution diversifies species of vertebrates. Further increase in oxygenation of atomosphere due to emergence of woody land plants. Pangea emerges. 
    
    + 200 million years ago: Dinosaurs! Followed by Permian extinction, loss of 95% species diversity. 
    
    + 60 million years ago: Evolution of mammals and other vertebrates. Dramatic global warming followed by ice age.
    
    + 2 million years ago - today: Evolution of humans and mammals post ice age. 


* Describe the dominant physical and chemical characteristics of Earth systems at the following waypoints:  

    + Hadea: Ball of molten atmosphere and molten crust.
    
    + Archean: Early oxgenation of the atmosphere with photosynthesis prokaryotes. High concentration of hydrocarbons in atmosphere. Sun was 30% less luminous. 
    
    + Precambrian: End of snowball Earth. Species see explosion in diversity. Atmosphere and geological properties support different metabolic processes and nutrient cycling at the most basic level. 
    
    + Proterozoic: Plants and animals differentiate, causing environmental conditions to support autotrophs and heterotrophs. Greater oxygenation of the atmosphere to support multicellular respiration. 
    
    + Phanerozoic: Global warming and cooling cycles between ice age and warming. Causing bloom in species diversity followed by mass extinctions. Atmosphere and geologic conditions become rich in organic nutrients quickly giving birth to new species when the niche is right. Emergence of complex carbon, oxygen, nitrogen, phosphorous, and sulfur cycling. 
    
-----------------

## Problem set 02
#### Learning objectives: 
Discuss	 the	 role	of	microbial	diversity and	 formation	of	coupled	metabolism	in	driving global	biogeochemical	cycles.

#### General questions:
* What	 are	 the	 primary	 geophysical	 and	 biogeochemical	 processes	 that	 create	 and	 sustain	conditions	for	life	on	Earth? How	do	abiotic	versus	biotic	processes vary	with	respect	to	matter and	energy	transformation and	how	are	they	interconnected?

(Answer): Volcanism and rock weathering and plate tectonics helped to shape the geological conditions for life. Supplies and sinks carbon dioxide and sulfur cycling.

* Why	is Earth’s	redox	state	considered	an emergent	property?

(Answer): Life on earth cannot be attributed to a single factor, therefore, the entire system ought to be examined as a whole. Where the whole is more than a sum of its parts. Many organisms needs to be coupled to other organisms where one biological process can only occur when linked to another.
  
* How	do reversible	electron transfer	reactions	give rise	to	element and nutrient cycles at different	ecological scales? What	strategies do	microbes use	to	overcome thermodynamic barriers to	reversible electron	flow?

(Answer): Thermodynamics determines favorable flow of electrons that allows biological reactions to occur spontaneously. Each reaction is spontaneous only in the environment which it occurs in, and in order to revert the reaction, the environmental conditions need to change for the reverse reaction to be spontaneous. This causes nutrients to be generally localized, preventing a global even distribution of nutrients in all environments. This builds a sort of "economy of scale" within microbial communities. The scale which thermodynamic events can occur is therefore dependent on the energetic favorability of certain organisms/communities of organisms to take place versus others. 

* Using information provided in the text, describe how the nitrogen cycle	partitions between	different redox “niches” and microbial groups. Is there a relationship between the nitrogen	cycle	and climate	change?	

(Answer): Microbes generally form distirbuted groups that carry out certain metabolic functions within their communities. These groups specialized into niches that carry out only one reaction within the nitrogen cycle. For example a group of organisms that reduces N2 gas to ammonium, and the reverse reaction of oxidizing ammonium are carried out by niches of other communities. Nitrous oxide is a potent greenhouse gas which can be pushed towards with Le Chatelier's principle on a population/community scale. 

* What is	the	relationship between	microbial	diversity	and	metabolic	diversity	and	how	does this relate	to	the	discovery	of new	protein	families from	microbial	community	genomes?

(Answer): Much of metabolic diversity arises from the adaptation of existing metabolic pathways due to great correlation and co-evolution between microbial species. Microorganisms utilize available resources from their environment in the most thermaldynamically favorable means, which often times result in modulation and altered implementation of existing pathways and enzymes, rather than evolving entirely novel mechanisms. This can be done over time through horizontal gene transfers between species, and emergence of new species from existing ones based on genetic differentiation. As species diversity increases, metabolic diversity increases logarithmically, meaning fewer and fewer new proteins without homologs are discovered for every species discovered.

* On	what	basis	do	the	authors	consider	microbes	the	guardians	of	metabolism?

(Answer): The Authors claim that due to horizontal gene transfer and the general conserved nature of genes that encode metabolic pathway enzymes, microbes that may survive a certain selection pressure can go on to preserve their metabolic capabilities and transfer them to the overall environment. Regardless of what the selection pressures are or how strong they are, certain species will always survive, and they serve as a sort of noah's ark for metabolic genes so that the environment may continue to cycle nutrients.

-----------------

## Working with Data in R (Data science Friday 4)
###_Tony Yan (18752148)_
###_version 1/31/2018_

### Exercise 1:
_We have another data table we will be using in up-coming DS Fridays, Saanich.OTU.txt. Import this data as a named object with the correct parameters in R._
```{r tables}
read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings=c("NAN", "NA", "."))
Saanich_metadata = read.table(file="Saanich.metadata.txt", header=TRUE, row.names=1, sep="\t", na.strings=c("NAN", "NA", "."))
```
### Exercise 2:
_Using dplyr, find at what depth(s) methane (CH4) is above 100 nM while temperature is below 10 °C. Print out a table showing only the depth, methane, and temperature data._
```{r dplyr}

library(dplyr)

select(Saanich_metadata, starts_with("CH4"))
filter(Saanich_metadata, CH4_nM > 100 & Temperature_C < 10) %>% select(Depth_m)
```
### Exercise 3:
_Convert all variables that are in nM to μM. Output a table showing only the original nM and converted μM variables._
```{r dplyr_tables}

library(dplyr)

select(Saanich_metadata, ends_with("nM")) %>%
  mutate(N2O_uM = N2O_nM/1000) %>%
  mutate(Std_N2O_uM = Std_N2O_nM/1000) %>%
  mutate(CH4_uM = CH4_nM/1000) %>%
  mutate(Std_CH4_uM = Std_CH4_nM/1000) 

```

----------------

##Plotting data in R (Data Science Friday 5)
###Tony Yan
###version February 08, 2018

### Exercise 1: Plot another nutrient of your choice against depth. Change the points to be purple triangles. 
```{r ggplot}

library(tidyverse)
library(phyloseq)

ggplot(Saanich_metadata, aes(x=NO3_uM, y=Depth_m, size=O2_uM)) +
  geom_point(color="purple", shape=17)
```

### Excercise 2: Using diplyr, convert temperature variable from Celcius to Fahrenheit. Then create a dot plot of temperature in Fahrenheit against depth.
```{r convert}
select(Saanich_metadata, Temperature_C, Depth_m) %>% 
  mutate(Temperature_F = (Temperature_C * (9/5)+32))

exercise2 = select(Saanich_metadata, Temperature_C, Depth_m) %>% 
  mutate(Temperature_F = (Temperature_C * (9/5)+32))

ggplot(exercise2, aes(x=Temperature_F, y=Depth_m)) +
  geom_point(color="red")
```

### Exercise 3: Create a bar plot at a different taxonomic level with more descriptive x- and y-axis labels as well as a title. Use the ggplot cheatsheet to learn the guides needed to change these labels. 
```{r barplot}
load("phyloseq_object.RData")
physeq_percent = transform_sample_counts(physeq, function(x) 100 * x/sum(x))
plot_bar(physeq_percent, fill="Class") +
    geom_bar(aes(fill=Class), stat="identity") +
  ggtitle("Taxonomic abundance from 10-200 m in Saanich Inlet") +
  xlab("Saanich Lake Sample Depth") +
  ylab("Percent abundance from total abundance")

plot_bar(physeq_percent, fill="Phylum") +
  geom_bar(aes(fill=Phylum), stat="identity") +
  facet_wrap(~Phylum, scales="free_y") +
  theme(legend.position="none")
```

### Exercise 4: Using ggplot, create a faceted figure showing nutrient concentrations in nM for O2, PO4, SiO2, NO3, NH4, and NO2 by depth. For this, you can use either the metadata table or the phyloseq object.
```{r faceted_figure}
exercise4 = select(Saanich_metadata, ends_with("uM"), Depth_m) %>%
  mutate(NH4_nM = NH4_uM*1000) %>%
  mutate(NO2_nM = NO2_uM*1000) %>%
  mutate(NO3_nM = NO3_uM*1000) %>%
  mutate(O2_nM = O2_uM*1000) %>%
  mutate(PO4_nM = PO4_uM*1000) %>%
  mutate(SiO2_nM = SiO2_uM*1000)

# Update: Concentrations were listed for "nM" in the assignment page, but later corrected to "uM". Exercise 4 was completed using the original instructions in "nM"

faceted = gather(exercise4, key = "Nutrient", value = "nM", ends_with("nM"))

ggplot(faceted, aes(x=Depth_m, y=nM))+
  geom_line()+
  geom_point()+
  facet_wrap(~Nutrient, scales="free_y") +
  theme(legend.position="none")

```

## Evidence worksheet_03 “The Anthropocene”
#### Learning objectives: 
*	Evaluate human impacts on the ecology and biogeochemistry of Earth systems.
General Questions:
* What were the main questions being asked?

(Answer): Rockstrom et al. focused on identifying the various limits of each global ecological "boundaries" and how far humans have come to those boundaries. These boundaries includes climate change (atmopheric greenhouse gas), biodiversity loss (rate), nitrogen cycle (artificial fixation), phosphorous cycle (P flowing into ocean), stratospheric ozone depletion, ocean acidification, global freshwater use, land use, and atmospheric/chemical pollution.  

* What were the primary methodological approaches used?

(Answer): Aggregate data based on multiple studies for each individual category was used to propose the boudnary level. The estimations were made conservatively, using rise-adverse approach. For example, greenhouse gas limit was estimated using atmospheric concentration and radiative forcing (rate of energy change per unit area). Other areas are estimated using comparisons of modern day measurements to historical fossil records to determine human impact. 

* Summarize the main results or findings. 

(Answer):
Proposed Boundaries
Climate change: 350 parts per million by volume
Biodiversity loss: 10 per million species per year
Nitrogen cycle: 35 millions of tonnes N2 per year removed from atmosphere
Phosphorus cycle: 11 millions of tonnes per year P flowing into ocean
Stratospheric ozone depletion: 276 Dobson unit
Ocean Acidification: 2.75 Global mean saturation state of argonite in surface sea water
Global freshwater use: 4000 km^3 per year
Change in land use: 15 Percent of global land cover used as cropland
Atmospheric arosol loading + Chemical polution: TBD

* Do new questions arise from the results?

(Answer): What is a "conservative estimate"? The paper does not indicate how the balance points are determined based on a consequence threshhold. The paper is limited in their approach to demonstrate what impacts of their proposed boundaries versus other proposed boundaries that exist in literature, and fail to demonstrate why theirs is methodologically more robust. 

* Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

(Answer): The Estimation methods were not greatly elaborated, rather their determination of the arbitrary threshhold, although based on aggregate metadata, does not show how the environment shifts in response to these boundaries, and why we should care about their proposed boundaries versus others. 

---------------

## Writing Assignment 1

Prompt: “Microbial life can easily live without us; we, however, cannot survive without the global catalysis and environmental transformations it provides”

Microbes have existed on Earth for eons, about 4 if one were to count. Their diverse evolutionary history is intimately intertwined with teraformation of the planet and its atmosphere, as well as its co-evolutionary relationship with most multicellular vertebrates that have, and will ever exist. The role of microbial life on the thriving flora and fauna of today is immense, but often overshadowed and overlooked by their larger and multicellular counterparts. At least in the eyes of the human conservationalists of today. Through the emergence of advanced molecular genetics techniques (such as genetic engineering and sequencing) and computational biology (bioinformatics) of recent times, humans have collectively just begun to realize (relative to anthropogenic time), and appreciate the biogeochemical role which microbes play to shape our world to be just the way it is. We have now uncovered evidence of early prokaryotic organisms harboring the genes and gene transfer mechanisms to serve as Earth’s Noah’s Ark, through mass extinctions and global biodiversity shifts in history [1]. 

Early evidence of horizontal gene transfer and our models of the emergent genes [1], we see that much of the metabolic and catalytic properties of microbes have been passed on from ancestors to descendants, from a time where the global metabolism is very different than they are today, yet these processes appear to be conserved. Relating that to the great oxygenation events of eons past, a clear trend comes into focus. Microbes shaped our planet; through their ability to carry out global catalysis of environmental nutrients/wastes in scales never thought possible. Figuratively speaking, they are ecologically “omnipotent”, existing in abundances that far outstrip all other living organisms combined. And they are omnipresent, rapidly evolving and filling ecological niches (physical or metabolic) nearly as quickly as they emerge. With rapid mutations and gene transfer capabilities, they are resilient, impossible to wipe out in their entirety [1]. It is foolish and hubris to believe, in any relevant capacity, that any form of intelligent life on Earth can come close to replacing the metabolic roles currently occupied by the microbes, and equally hubris to assume that life on Earth can exist without them. 

But as much as microbes possess qualities that can only be anthropologically compared to gods and deities, microbes need us. They need humans, perhaps now more than ever before; perhaps as much as we need them. Ecologically speaking, times are changing. With humans rapidly industrializing by the minute; evidence shows that humans have significantly altered the Earth’s delicate biogeochemical cycles, such as the nitrogen and carbon cycle, in ways that can no longer be balanced by microbes in their current state [4]. Humans have accomplished this through a timeframe that is ecologically unheard of, and is doing so exponentially. This has been demonstrated to have caused severe, quantifiable, and irreversible biodiversity loss [4]. Simple as it is to substantiate the claim, “humans are unable to survive without the environmental transformation microbes provide”, the reverse is true for the subsequent claim: “Microbial life can easily live without us”, in the status quo.

Microbial life’s greatest strength is largely warranted by its sheer magnitude/abundance to which they can exist. Whitman et al. estimates in “Prokaryotes: the unseen majority” there are 4-6 x 1030 prokaryotic organisms dispersed among general environmental groups: marine, soil, subsurface, and microbiota environments [3]; and their respective carbon cycle participation is anywhere between matching to doubling that of plants and animals [3]. Their total role as nitrogen cycle and phosphorous cycle reservoirs is estimated to be about 10-folds greater than all other living organisms [3]. The massive scale which prokaryotic metabolism operates affords relative stability to perturbations from the environment. “Microbial engines” as modeled by Falkowski and colleagues [1], are relatively robust. However, as with any intricate system, significant shifts or adjustments will destabilize the tightly coevolved microbial engine. Rockstrom argues that modern industrial agricultural processes such as the synthetic fixation of gaseous nitrogen to reduced forms in fertilizer, is causing a multiple-fold increase in nitrogen fixation in the environment, causing destabilization of large marine-environments (eutrophication, anoxia) beyond the ability for the environment to re-stabilize [2]. Rockstrom et al. and Canfield et al. both observed that the amount of industrially fixed nitrogen more than matches all terrestrial processes, contributing significantly to global greenhouse gas emissions, pushing for the formation of nitrous oxide (a potent greenhouse gas), through ecological processes not unlike Le Chatelier’s principle [3, 4]. The phosphorous cycle has experienced a similar phenomenon, where Earth’s natural sediment reservoir is being uplifted by human industry and fed back into the biological phosphorous cycle; that if continued, will lead to anoxia in seas and oceans [3]. Ultimately, long term destabilization of ecological systems will result in large-scale environmental shifts if not corrected for, an extension of the climate change we observe today.

Keen proponents of microbial ecosystem resilience, to artificial influence, would be quick to remark upon the unique prokaryotic capabilities for rapid genetic modification, through the means of direct genetic mutation as well as horizontal gene transfer. Falkowski et al. indeed characterizes the relationship between environmental processes and the microbial genome as a metaphorical “microbial engine”. Falkowski et al. describes, based on molecular evidence (gene order and metabolic distribution), that much of today’s biogeochemical cycles emerged from evolutionary time, through the means of horizontal propagation, which lead to a wide distribution of metabolic genes between events of cellular differentiation [1, 7]. Today we can observe a wide taxonomic range of organisms favoring the sharing of an increasingly conserved pool of global metabolic genes, over the evolution of brand new genes with no known homologs, or at least the ones we know to exist [Falkowski et al, 2008: fig. 3]. The concept of “Microbes as guardians of metabolism” does not describe rapid genetic modifications as a mechanism for increasing short term fitness, but rather for long term evolutionary survival on a population scale. Population genomics through horizontal gene transfer is no doubt capable of maintaining key metabolic machinery in long term within select organisms, much like a Noah’s Ark as an analogy. However, little evidence exists to show that these genetic modulation mechanisms are capable of halting/slowing down biodiversity loss as due to a shift in selection pressure. Even with consideration of the “Microbes as guardians of metabolism” concept, the collective consensus within the scientific community points to the ultimate loss of much of Earth’s microbial diversity seen in the status quo with respect to anthropogenic climate change. 

Earth’s evolutionary history operates in an entirely different timeframe than our anthropogenic history. While the human legacy spans a respectable 200,000 years since the homo sapiens species origin, microbial life has been shaping the fate of Earth since approximately 4 billion years ago. Through this time, numerous rapid environmental changes occurred, which resulted in emergence of new eras of global microbial ecosystems characterized by their impact on the biogeochemical properties of the planet. Using oxygenation of the atmosphere as an example of evolutionary timeframes, Nisbet and Sleep describes early microbial history to be “bottlenecked” by the relative inhospitable conditions that would rapidly arise. Nisbet and Sleep attributes early bottlenecks of life to the relatively low sunlight followed by volcanic and extraterrestrial bombardments [5]. The emergence of oxygenation from the carbon dioxide and ammonium rich early Earth atmosphere had only been accomplished through the evolution of several layers of increasingly specialized phototrophic microorganisms over the course of eons [4, 5], both prokaryotic and eventually eukaryotic. The primary engine component attributed to for the oxygenation of Earth’s atmospheric oxygen composition is the emergence of cyanobacteria. In its current form, it’s emergence was characterized to 2.3 billion years ago, nearly two eons prior to the first emergence of phototrophy [6]. From that point, numerous eras furthered the atmospheric oxygenation needed to sustain vertebrate life. It is safe to say; the Earth will not be repeating the biogeochemical timeframes it once took to deliver humans the world which we live on today. 

For better or for worse, humans have dramatically altered the world in the course of a mere handful of milliseconds following the evolutionary clock analogy. Chris Mooney from the Washington Post article titled: “Scientists say humans have now brought on an entirely new geologic epoch”, claims that the combination of atmospheric modulation and sedimentary disturbances followed by widespread flora and fauna disruption, from human industrial activity, has left enough of impact to become detected in geological history. This warrants the ushering of the “Anthropocene” the era of humans. As evident by the stark contrast in timeframes which the Earth’s microbial engines operates versus humans, the question no longer remains whether microbial engines are capable of driving the planet forward in a manner that sustains our large planetary biodiversity, microbial or otherwise. The new question emerges, whether humans have any role to play in this new global paradigm. 

Continuing with the train of metaphors, Joel Achenbach from the Washington Post article titled: “Spaceship Earth: A new view of metaphors” portrays our planetary metabolic network as a “Spacehship Earth, enter[ing] 2012 belching smoke, overheating and burning through fuel at a frightening rate. It’s feeling pretty crowded, and the crew is mutinous. No one’s at the helm”. Summarizing the reality at hand; humans, through industrialization, has pushed our environmental metabolic network beyond its limits for destabilization; warranted by the introduction of industrially fixed nitrogen and overturned sedimentary phosphorous species among many more. Ultimately resulting in a change in selection pressure, that leads to the shift of ecological conditions away from the status quo and cements a future that is less habitable by the world’s current biodiversity. 

While it is true that microbial life, through horizontal gene transfers, exhibits high metabolic resiliency on the community level, individuals and entire species are put at risk due to environmental metabolic shifting. Microbial metabolic potential will carry on in small populations of microorganisms (Noah’s Ark analogy), but the vast majority of biodiversity will become extinguished in the event of ecological catastrophes. And the impact of this includes a reduced potential for Earth to sustain life, microbial, human, or otherwise, moving forward into the future. 

Whether humans can survive on a planet without microbial catalysis, or whether metabolic potential within small fringe microbial communities can survive through a given human-induced ecological catastrophe, both are fruitless debates. In an ideal “Spaceship Earth”, there ought to be a captain, crew, and engines to keep it all going. Falkowski’s “Microbial engines” represents the baseline catalysis of metabolites by microbial genomic potential, but ultimately requires human intervention, a “captain at the helm” to offset evolutionary forces acting on a human destabilized ecosystem. Joel Achenbach outlines how the time for interventionist ecology is now, and that humans are in a better position technologically and logistically than ever before to offset the industrial metabolites that are being offloaded into the natural nutrient cycles [2, 4]. Humans are capable of renewable energy, biochemical/atmospheric engineering, and molecular genetics techniques all readily implementable to tackle the ecological damage in a directed and stepwise manner. Humans are becoming smarter, faster, and more just, at least according to Joel from “Spaceship Earth”. Socioeconomic infrastructure is developing alongside technological ones to promote sustainable growth, responsible development, and a more wholly encompassing worldview that spans beyond the human race. Maybe, just maybe, there can exist a future where intelligent life co-exists with microbial catalysis on the global scale that promotes symbiosis between mother nature, and sentient industriousness. 

References:

1.	Falkowski PG, Fenchel T, Delong EF. 2008. The Microbial Engines That Drive Earth’s Biological Cycles. Science 320:1034-1038. 
2.	Rocksrtrom J, et al. 2009. A safe operating space for humanity. Nature 461:472-475
3.	Whitman WB, Coleman DC, Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc. Natl. Acad. Sci. 95:6578-6583. 
4.	Canfield DE, Glazer AN, Falkowski PG. 2010. The Evolution and Future of Earth’s Nitrogen Cycle. Science 330:192-196.
5.	Nisbet EG, Sleep NH. 2001. The habitat and nature of early life. Nature 409:1083-1091.
6.	Kasting JF, Siefert JL. 2003. Life and the Evolution of Earth’s Atmosphere. Science 296:1066-1067. 
7.	Falkowski PG et al. 2000. The Global Carbon Cycle. Science 290:291-296. 

----------------

### Module 01 references
Utilize this space to include a bibliography of any literature you want associated with this module. We recommend keeping this as the final header under each module.

Whitman WB, Coleman DC, and Wiebe WJ. 1998. Prokaryotes: The unseen majority. Proc Natl Acad Sci USA. 95(12):6578–6583.  [PMC33863](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC33863/)

-----------------

# Module 02

## Module 02 portfolio check
The first of your second level headers (##) is to be used for the portfolio content checks. The Module 01 portfolio check has been built for you directly into this template, but will also be available as a stand-alone markdown document available on the MICB425 GitHub so that you know what is required in each module section in your portfolio. The completion status and comments will be filled in by the instructors during portfolio checks when your current portfolios are pulled from GitHub.

* Evidence worksheet_04
    + Completion status:
    + Comments:
* Problem Set_03
    + Completion status:
    + Comments:
* Additional Readings
    + Completion status:
    + Comments 

### Evidence Worksheet_04
### Learning objectives:
* Discuss the relationship between microbial community structure and metabolic diversity
* Evaluate common methods for studying the diversity of microbial communities
* Recognize basic design elements in metagenomic workflows

### General Questions:
* What were the main questions being asked?

(Answer): To understand to a greater genetics and biochemical degree regarding the functional capacity of proteorhodopsins and their effects on the microbial metabolic potential and their ability to enhance growth when transferred via HGT.

* What were the primary methodological approaches used?

(Answer): Functional screening through large insertion library of clones expressing PR photosystems. Transposon mutants were analyzed for biochemical products of the PR genes. Heterologous expression experiments were used to show that the PR genes can be aquired through HGT and that phototrophy is a transient property.

* Summarize the main results or findings. 

(Answer): Functional screens of metagenomic libraries without cloning was able to achieve higher levels of recombinant gene expression into detectable phenotypes. Using these methods, they identified and developed a method to demonstrate that PR genes can be transferred and heterologously expressed, while to an extent that can substantially alter the growth rates of organisms that uptake this system. 

* Do new questions arise from the results?

(Answer): How do bacteria gather these genes and transfer them in reality? (AKA, not in a controlled test environment where the mechanism is determined as a manipulated variable)? Is there evidence of enhanced growth in PR capable species in nature by such things as abundance measurements in recovered sequencing reads?

* Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

(Answer): One of the biggest assumptions made is that ATP synthesis was measured as a proxy for the growth rates. It is unclear/incomplete that the author is using a proxy for growth rather than measuring the growth exactly. Building more ATP does not represent a true measure for growth since having more temporary energy sources does not necessarily dictate the growth of an organism in the actual environment. 

-------------

## Problem set_03 “Metagenomics: Genomic Analysis of Microbial Communities”
#### Learning objectives: 
Specific emphasis should be placed on the process used to find the answer. Be as comprehensive as possible e.g. provide URLs for web sources, literature citations, etc.  
*(Reminders for how to format links, etc in RMarkdown are in the RMarkdown Cheat Sheets)*

#### Specific Questions:
* How many prokaryotic divisions have been described and how many have no cultured representatives (microbial dark matter)?

(Answer): As of 2003, 26 of the 52 major bacterial phyla have been cultured, only about 50%. The estimate is far fewer now, with more uncultured species being discovered. In 2016, only 89 bactetial phyla and 20 archeal phyla (examined via the 16s rRNA databases). All of this amounts to approximately 1% of microbes in a given ecological niche is cultivatable.   

* How many metagenome sequencing projects are currently available in the public domain and what types of environments are they sourced from?

(Answer): Thousands are available in the public domain, sourcing from a wide range of environments from marine to soil environments. With microbiota, marine, sediments, and aquatic being some of the more prevalent environments. These projects shed light on difficult to culture/uncultured organisms from those environments.  

* What types of on-line resources are available for warehousing and/or analyzing environmental sequence information (provide names, URLS and applications)?  

(Answer): 
Oulas, Anastasis, et al. 2015. Metagenomics: Tools and insights for analyzing next-generation sequencing data derived from biodiversity studies. Bioinformatics and biology insights 9:BBI-S12462.

Siegwald, Lea, et al. 2017. Assessment of common and emerging bioinformatics pipelines for targeted metagenomics. PLoS One 12.1:e0169563.

Assembly: Euler, MEGAHIT, VELVET
Binning: MaxBin, S-GCOM
Annotation: KEGG, COG, Metacyc, SILVA
Analysis Pipelines: Megan5
Standalone Software: OTUbase
Denoising: AmpliconNoise
Databses: Ribosomal Database Project, Refseq, SILVA

* What is the difference between phylogenetic and functional gene anchors and how can they be used in metagenome analysis?   

(Answer): Phylogenetic gene anchors differ from functional ones in that phylogenetic anchors represent conserved regions of essential genes that can be a base of comparison between most/all organsims. Such genes do not undergo divergent mutations due to their necessity in sustaining biological life. Functional gene anchors are a base of comparison not between organisms themselves, but between functions and metabolic steps that is investigated. Will show the convergence/divergence of a metabolic process as a whole rather than the composition of the community within. 

* What is metagenomic sequence binning? What types of algorithmic approaches are used to produce sequence bins? What are some risks and opportunities associated with using sequence bins for metabolic reconstruction of uncultivated microorganisms?  

(Answer): Binning is the process of grouping sequences in the aim of reconstructing the genome of a single organism from a large set of genome data. 
Algorithm:
1. Align sequence to database (guided binning)
2. Grouping based on DNA characteristics such as GC content (denovo)
Risks: Incomplete coverage of the genome sequence (very likely), or contamination from different phylogeny. 

* Is there an alternative to metagenomic shotgun sequencing that can be used to access the metabolic potential of uncultivated microorganisms? What are some risks and opportunities associated with this alternative?  

(Answer): Single cell genomics through cell sorting (with FACs or the likes), and reconstructing the genomes of single cells is the emerging field that contrasts with WGS. The advantages is that single cell genomes are much less likely to suffer from high phylogenetic contamination, but due to the sampling method (sampling bias), the coverage of the genome may not be very good, yielding low completeness. 

-----------------

# Module 03

## Module 03 Portfolio Content

* Evidence worksheet_05
    + Completion status:
    + Comments:
* Problem set_04
    + Completion status:
    + Comments:
* Writing Assessment_03
    + Completion status:
    + Comments:
* Project_1
    + Completion status:
    + Comments:
    
## Problem set_04 “Fine-scale phylogenetic architecture”
#### Learning objectives: 

* Gain experience estimating diversity within a hypothetical microbial community

#### Part 1: Description and enumeration
Obtain a collection of "microbial" cells from "seawater". The cells were concentrated from different depth intervals by a marine microbiologist travelling along the Line-P transect in the northeast subarctic Pacific Ocean off the coast of Vancouver Island British Columbia. 

Sort out and identify different microbial "species" based on shared properties or traits. Record your data in this Rmarkdown using the example data as a guide.

Once you have defined your binning criteria, separate the cells using the sampling bags provided. These operational taxonomic units (OTUs) will be considered separate "species". This problem set is based on content available at [What is Biodiversity](http://cnx.org/content/m12179/latest/).

For example, load in the packages you will use.
```{r message=FALSE}
#To make tables
library(kableExtra)
library(knitr)
#To manipulate and plot data
library(tidyverse)
```

Then load in the data. You should use a similar format to record your community data.
```{r}
example_data1 = data.frame(
  number = c(1,2,3),
  name = c("lion", "tiger", "bear"),
  characteristics = c("brown cat", "striped cat", "not a cat"),
  occurences = c(2, 4, 1)
)

```

Finally, use these data to create a table.
```{r}
example_data1 %>% 
  kable("html") %>%
  kable_styling(bootstrap_options = "striped", font_size = 10, full_width = F)

```
	
For your sample:

* Construct a table listing each species, its distinguishing characteristics, the name you have given it, and the number of occurrences of the species in the collection.
* Ask yourself if your collection of microbial cells from seawater represents the actual diversity of microorganisms inhabiting waters along the Line-P transect. Were the majority of different species sampled or were many missed? 

```{r part1}

candy_counts = read.table(file="Candy_counts.csv", header=TRUE, sep=",", na.strings=c("NAN", "NA", "."))
candy_table = candy_counts %>% kable("html")

# The collection of microbial cells from seawater missed many species, especially if they were already in low abundance. 

```
#### Part 2: Collector's curve
To help answer the questions raised in Part 1, you will conduct a simple but informative analysis that is a standard practice in biodiversity surveys. This analysis involves constructing a collector's curve that plots the cumulative number of species observed along the y-axis and the cumulative number of individuals classified along the x-axis. This curve is an increasing function with a slope that will decrease as more individuals are classified and as fewer species remain to be identified. If sampling stops while the curve is still rapidly increasing then this indicates that sampling is incomplete and many species remain undetected. Alternatively, if the slope of the curve reaches zero (flattens out), sampling is likely more than adequate.

To construct the curve for your samples, choose a cell within the collection at random. This will be your first data point, such that X = 1 and Y = 1. Next, move consistently in any direction to a new cell and record whether it is different from the first. In this step X = 2, but Y may remain 1 or change to 2 if the individual represents a new species. Repeat this process until you have proceeded through all cells in your collection.

For example, we load in these data.
```{r example}
example_data2 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10),
  y = c(1,2,3,4,4,5,5,5,6,6)
)
```

And then create a plot. We will use a scatterplot (geom_point) to plot the raw data and then add a smoother to see the overall trend of the data.
```{r example_graph}
ggplot(example_data2, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```

For your sample:

* Create a collector's curve.
* Does the curve flatten out? If so, after how many individual cells have been collected?
* What can you conclude from the shape of your collector's curve as to your depth of sampling?

```{r part2}

candy_count_collector = data.frame(
  x = c(5,6,7,10,20,28,34,41,44,46,47,50,53,59,60,61,62,63,64,65,66,75,79,84,91,97,102,106,112,129,142,145,146,147),
  y = c(1,2,3,4,4,4,4,4,5,5,5,5,5,5,6,7,7,8,9,9,9,10,10,10,10,10,11,11,11,11,11,11,12,13)
)

ggplot(candy_count_collector, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")

# The curve flattens out at around 12 species, the total number of individuals collected is 147.
# In the beginning and towards the end, sampling individuals doesn't increase the species much, but in the middle, sampling more individuals is corresponded with a rapid increase of species.
```
#### Part 3: Diversity estimates (alpha diversity)
Using the table from Part 1, calculate species diversity using the following indices or metrics.

##### Diversity: Simpson Reciprocal Index

$\frac{1}{D}$ where $D = \sum p_i^2$

$p_i$ = the fractional abundance of the $i^{th}$ species

For example, using the example data 1 with 3 species with 2, 4, and 1 individuals each, D =

```{r}
species1 = 2/(2+4+1)
species2 = 4/(2+4+1)
species3 = 1/(2+4+1)

1 / (species1^2 + species2^2 + species3^2)


Kiss = 5/147
Red_candy = 1/147
Red_long_candy = 1/147
Mike_ike = 34/147
Gummy_bear = 18/147
Lego_square = 1/147
Lego = 2/147
Spider = 1/147
Big_round = 3/147
Skittle = 31/147
MM = 48/147
Fish = 1/147
cloud = 1/147

Simpsons = 1/ (Kiss^2 + Red_candy^2 + Red_long_candy^2 + Mike_ike^2 + Gummy_bear^2 + Lego_square^2 + Lego^2 + Spider^2 + Big_round^2 + Skittle^2 + MM^2 + Fish^2 + cloud^2)

```

The higher the value is, the greater the diversity. The maximum value is the number of species in the sample, which occurs when all species contain an equal number of individuals. Because the index reflects the number of species present (richness) and the relative proportions of each species with a community (evenness), this metric is a diveristy metric. Consider that a community can have the same number of species (equal richness) but manifest a skewed distribution in the proportion of each species (unequal evenness), which would result in different diveristy values.

* What is the Simpson Reciprocal Index for your sample?

##### Richness: Chao1 richness estimator
Another way to calculate diversity is to estimate the number of species that are present in a sample based on the empirical data to give an upper boundary of the richness of a sample. Here, we use the Chao1 richness estimator. 

$S_{chao1} = S_{obs} + \frac{a^2}{2b})$

$S_{obs}$ = total number of species observed
a = species observed once
b = species observed twice or more

So for our previous example community of 3 species with 2, 4, and 1 individuals each, $S_{chao1}$ =

```{r}
3 + 1^2/(2*2)

Chao1 = (13 + ((6^2)/(2*7)))
```

* What is the chao1 estimate for your sample?

#### Part 4: Alpha-diversity functions in R
We've been doing the above calculations by hand, which is a very good exercise to aid in understanding the math behind these estimates. Not surprisingly, these same calculations can be done with R functions. Since we just have a species table, we will use the `vegan` package. You will need to install this package if you have not done so previously. 

```{r message=FALSE}
library(vegan)
```

First, we must remove the unnecesary data columns and transpose the data so that `vegan` reads it as a species table with species as columns and rows as samples (of which you only have 1).
```{r}
example_data1_diversity = 
  example_data1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

example_data1_diversity

candy2 = read.table(file="candy2.csv", header=TRUE, sep=",", na.strings=c("NAN", "NA", "."))

candy_diversity =
  candy2 %>%
  select(Name, Count) %>%
  spread(Name, Count)
```

Then we can calculate the Simpson Reciprocal Index using the `diversity` function.
```{r}
diversity(example_data1_diversity, index="invsimpson")

diversity(candy_diversity, index="invsimpson")
```

And we can calculate the Chao1 richness estimator (and others by default) with the the `specpool` function for extrapolated species richness. This function rounds to the nearest whole number so the value will be slightly different that what you've calculated above.
```{r}
specpool(example_data1_diversity)

specpool(candy_diversity)
```

In Project 1, you will also see functions for calculating alpha-diversity in the `phyloseq` package since we will be working with data in that form.

For your sample:

* What is the Simpson Reciprocal Index using the R function?
* What is the chao1 estimate using the R function?
    + Verify that these values match your previous calculations.

```{r}
Community_total1 = data.frame(
  number = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20),
  name = c("kiss", "red short string", "red long string", "white spiral", "mike ike", "watermelon", "gummy bear", "marshmallow bear", "lego square", "lego rectangle", "coke", "spider", "big round candy", "skittle", "m&m", "fish", "elongated noodle", "elongated cloud", "elongated oval", "elongated diamond"),
  characteristics = c("teardrop shape", "short string", "long string", "large ellipsoid flat", "rounded rod", "rectangular", "clear bear shape", "opaque bear shape", "large square brick", "small rectangular brick", "coke bottle shape", "spider shape", "sphere", "ellipsoid round", "small elipsoid flat", "fish shape", "short noodle shape", "cloud shape", "oval", "diamond"),
  occurences = c(16,6,10,3,172,1,99,3,3,15,3,6,24,187,243,1,4,3,1,1)
)
```

Part 2: Collector's Curve
```{r}
Community_sample1 = data.frame(
  x = c(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20),
  y = c(0,2,6,7,41,41,57,57,57,62,62,64,70,110,164,164,164,164,165,165)
)
```

```{r}
ggplot(Community_sample1, aes(x=x, y=y)) +
  geom_point() +
  geom_smooth() +
  labs(x="Cumulative number of individuals classified", y="Cumulative number of species observed")
```

Part 3: Diversity estimates (alpha diversity)
Simpson Reciprocal Index (Community)
```{r}
species1 = 16/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species2 = 6/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species3 = 10/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species4 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species5 = 172/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species6 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species7 = 99/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species8 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species9 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species10 = 15/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species11 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species12 = 6/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species13 = 24/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species14 = 187/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species15 = 243/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species16 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species17 = 4/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species18 = 3/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species19 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)
species20 = 1/(16+6+10+3+172+1+99+3+3+15+3+6+24+187+243+1+4+3+1+1)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2 + species15^2 + species16^2 + species17^2 + species18^2 + species19^2 + species20^2)
```

Simpson Reciprocal Index (Sample)
```{r}
species1 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species2 = 2/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species3 = 4/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species4 = 1/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species5 = 34/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species6 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species7 = 16/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species8 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species9 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species10 = 5/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species11 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species12 = 2/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species13 = 6/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species14 = 30/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species15 = 54/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species16 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species17 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species18 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species19 = 1/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)
species20 = 0/(0+2+4+1+34+0+16+0+0+5+0+2+6+40+54+0+0+0+1+0)

1 / (species1^2 + species2^2 + species3^2 + species4^2 + species5^2 + species6^2 + species7^2 + species8^2 + species9^2 + species10^2 + species11^2 + species12^2 + species13^2 + species14^2 + species15^2 + species16^2 + species17^2 + species18^2 + species19^2 + species20^2)
```

Chao1 Richness Estimator (Community)
```{r}
20 + 4^2/(16*2)
```

Chao1 Richness Estimator (Sample)
```{r}
11 + 2^2/(9*2)
```

Part 4: Alpha-diversity functions in R
```{r}
library(vegan)

Community_total1_diversity = 
  Community_total1 %>% 
  select(name, occurences) %>% 
  spread(name, occurences)

Community_total1_diversity

diversity(Community_total1_diversity, index="invsimpson")
specpool(Community_total1_diversity)
```


#### Part 5: Concluding activity
If you are stuck on some of these final questions, reading the [Kunin et al. 2010](https://www.ncbi.nlm.nih.gov/pubmed/19725865) and [Lundin et al. 2012](https://www.ncbi.nlm.nih.gov/pubmed/23760801) papers may provide helpful insights. 

* How does the measure of diversity depend on the definition of species in your samples?
(Answer): Depending on what characteristeristics we choose to group individuals, a species is simply a category that can be bigger or smaller depending on what we choose to put inside it. 
* Can you think of alternative ways to cluster or bin your data that might change the observed number of species?
(Answer): Change the characteristics that we choose to cluster together as species (such as by color instead of by candy brand).
* How might different sequencing technologies influence observed diversity in a sample?
(Answer): Different sequencing types require different post-sequencing quality controls and assembly criterion which introduces different levels of sequencing error, and their corresponding computational biases depending on how the sequences are processed. 

### Evidence Worksheet_04
### Learning objectives:
* Discuss the relationship between microbial community structure and metabolic diversity
* Evaluate common methods for studying the diversity of microbial communities
* Recognize basic design elements in metagenomic workflows

### General Questions:
* What were the main questions being asked?

(Answer): What are some of the differences between pathogenic strains of E.coli and non-pathogenic strains on the genome level? What genes and genomic constructs differentiates them and gives certain strains their pathogenic qualities? Are these qualities transient?

* What were the primary methodological approaches used?

(Answer): Basic sequence annotation through annotating ORFs from sequences and BLAST hits were annotated to the ORFs and the genomes were compared after assembly. 

* Summarize the main results or findings. 

(Answer): They found that the primary genes responsible for pathogenecity is a type III secretion system and its corresponding toxin. Other notable features differentiating between pathogenecity and docility are presence of iron-sequestering systems, adhesins, autotransporters, and recombinases. These exist in pathogenecity islands which disrupts the paradigm that these genes were vertically evolved. 

* Do new questions arise from the results?

(Answer): They compared the genetic traits of the pathogenecity islands to determine the likelihood of having been inherited horizontally. If they were to have done the same with the host genomes, wouldn't they find substantial variation wthin the sequences? If so, how can they determine if something was not a part of the genome inherited vertically if they have no explicit control?

* Were there any specific challenges or advantages in understanding the paper (e.g. did the authors provide sufficient background information to understand experimental logic, were methods explained adequately, were any specific assumptions made, were conclusions justified based on the evidence, were the figures or tables useful and easy to understand)? 

(Answer): With the limit of the databse, it is hard to address where the pathogenic islands have come from. The authors did not do a screen through characterized phages or other bacterial species. It is confusing how they laid out figure 3, it is unclear which part of the genome that they are referring to, and how one vertical line can be represented as 100kbp. 

### Part 2: Learning objectives:
* Comment on the creative tension between gene loss, duplication and aquisition as it relates to microbial genome evolution.

(Answer): Horizontal gene evolution convolutes what we understand about vertical inheritance since we have no explicit controls of genomes that have never been evolutionarily touched by HGT. But due to this phenomenon, we see transient genes in places where they we would normally expect. To understand evolution, HGT and VGT need to be considered together, despite our best efforts to denoise HGT while studying lineages of genes/organisms. 

* Identify common molecular signatures used to infer genomic identity and cohesion.

(Answer): Genomic identity within a genome can be inferred with codon biases, GC content, and relative gene location that may give a clue to whether the genes existed within an organism to an extent long enough to have observably co-evolved with other genes in the organism.

* Differentiate between mobile elements and different modes of gene transfer. 

(Answer): Phages, Transposons, Plasmids, Transformation (conjugations), Recombination. Mobile genetic elements include plasmids and phage-packaged DNA. Mobile genetic elements are elements able to replicate themselves and are innately transient, while other types of gene transfer may rely on mechanisms by the cells or phage to be transient. 

Based on your reading and discussion notes, explain the meaning and content of the following figure derived from the comparative genomic analysis of three E. coli genomes by Welch et al. Remember that CFT073 is a uropathogenic strain and that EDL933 is an enterohemorrhagic strain. Explain how this study relates to your understanding of ecotype diversity. Provide a definition of ecotype in the context of the human body. Explain why certain subsets of genes in CFT073 provide adaptive traits under your ecological model and speculate on their mode of vertical descent or gene transfer.

(Answer): Comparing the two phenotypically distinct strains, we see that their pathogenecity islands are relatively similar with similar genes occuplying locus that are similar as well. They share a number of genes and even for genes that are different, occupy similar locational patterns also. This is evidence that these PAIs originated in some common ancestor and was somehow either horizontally or vertically distributed to these strains. With otherwise similar genomes, these "ecotypes" of bacteria are represent "distinct" forms of their species that occupy an ecological niche within a given environment. These "distinct" forms are highly subjective and in this case, can even fall upon transient properties, allowing the bacteria to gain or lose an "ecotype" based on HGT. 


## Writing Assignment 3

	When Charles Darwin published his theories on evolution and natural selection being driving forces for biological change in his book: “On the Origin of Species”, the concept of “microbial ecology” was but in its infancy. At that time, the concept of “species”, in other words, a unit of taxonomic similarity to which a group of organisms may collectively share a number of attributes, was only rudimentarily defined. The primary operational definition of a species, or a taxonomic group in general, consisted of phenotypical or sometimes behavioral characteristics of organisms that marked their similarity, or lack thereof. It is through these empirically derived observations on organisms that first emerged the concept of a “species”. Darwin’s classical school of thought served well into the 20th century, when the concept of taxonomy was sufficient to characterize and categorize much of the observable life on the planet. However, as technology grew, so did our ability to observe the previously unobservable. This manifested itself through the growth of microbiology and microbial ecology. As an increasing number of organisms become cultured and studied, the traditional rules, binding taxonomic groups together based on phenotypical similarity, begins to show its weakness and obsolescence. Furthermore, the introduction of genome sequencing punched the final hole in Darwin’s sinking ship. It became apparent that phenotypical similarity is no longer sufficient to describe the similarity between organisms, and even less so for inferring evolutionary history. 

	In the new age of molecular genetics, high throughput genome sequencing, and bioinformatics tools, observing microbial phenotypes is no longer a feasible nor indicative means of understanding relativeness of microorganisms. Understanding the properties of the genome has led us to understand that the phenotypical characteristics of organisms is in-fact the result of complicated series of mechanisms that regulate, express, exchange, and inherit genes within their respective genomes. Natural selection, long thought to be the driving force for observable phenotype change, was actually acting on a molecular scale; altering and selecting combinations of nucleotides. Thus, the minute similarities and differences between genes and genomes proved to be a better representation of microbial species and to represent their evolutionary heritage. Genetics was not only a paradigm shift for the microbial species concept, but also penetrated on biologists’ perception on eukaryotic, multicellular organisms, and even into the long-regarded traditionalist: botanical and zoological domains of life. 

	Earliest attempts to use genomics for characterizing microbial species came with the seemingly obvious approach, to directly compare one genome with another. Early genetics held the school of thought that if evolution acted on all life by applying selection pressure on the genes and genomes; therefore, the characterization and relatedness of species can be directly inferred from genes, and genome-wide differences. Early genetics was primarily bottlenecked by its ability to resolve small sequence differences, a combination of low-throughput sequencing (first generation Sanger sequencing), and amplification-based (PCR-based) genome qualification methods, followed by the bare-existence of genome databases. Back in the late 20th century, genome-wide comparisons were tremendously difficult to make between microorganisms. As new sequencing technology hit the mainstream, our genome repositories and capacity to resolve genomes grew exponentially. With it, came the rapid realization that microbial genomes are incredibly diverse, even within single samples or single organisms. The discovery of horizontal gene transfer (HGT) and mobile genetic elements as well as a better understanding of phage mechanics demonstrated that a microbial genome more closely resembled a patchwork quilt, than the clean slate that was previously thought to be true. These inconsistencies within even single genomes demonstrated a fundamental flaw in overreliance on genetics, as well as our previous concept of microbial species.

	An early attempt to characterize pathogenic and nonpathogenic strains of E. coli by R.A Welch and his team revealed just how diverse genome qualities are, even within a single well-learned species. Welch et al. 2002 compared the whole genomes of 3 strains of E coli. One strain being nonpathogenic while the other two are enterohemorrhagic and uropathogenic respectively. Welch found that between the three strains of a single E. coli species, their genomes only shared about 39% of expressed genes, and each strain is approximately equally different from each other, despite two pathogens having some phenotypes that are common among pathogenic strains (1). Furthermore, Welch found that even within pathogenic strains, the uropathogenic strain had many different pathogenicity factors that were not present within the enterohemorrhagic strain. The uropathogenic strain was further compared with two other uropathogenic E. coli strain genomes from database; and even between uropathogenic strains, Welch’s team observed that the main pathogenicity islands were largely variant (1). It showed signs of having been inherited from multiple independent clonal lineages. This shows that although the inferred vertically inherited genes are similar between strains, great portions of the backbone are disrupted by gene islands that are the result of HGT (1). 

	Through the popularization of high-throughput sequencing such as pyroseq and illumina, more robust methods/protocols have been adopted to characterize microbes through their genomes. At this point, the field collectively realized the enormous phylogenetic diversity and problems associated with whole-genome phylogenetic analysis. As exemplified through the methods used by Sogin’s team in their study of microbial diversity in deep sea biosphere, sequence clustering to select “marker” genes became the new standard. Sogin et al. took environmental sequences from a deep-sea sample and amplified the 16s rDNA sequences, then bioinformatically clustered the rDNA based on sequence similarity to form clusters of sequences that represent a single taxonomic unit (OTU) (2). 16s rDNA is used as amplification target due to its omnipresence in all domains of life, hence a “marker” that bridges the gap between vastly diverse genomes. Each single OTU was then assigned a taxonomic identity based on the OTU's similarity to database sequences (using BLAST local alignment, and MUSCLE multiple sequence alignment). The resulting OTUs represent known phylogenetic units on the tree of life. Sogin then analyzed these OTUs with respect to the sample richness and diversity of species through ACE and Chao1 estimates (2). The abundance of sequenced reads that map back to the OTUs serve as proxy for estimating the abundance of those organisms in the depth profile. 

	Clustering by 16s rRNA/rDNA along with other marker genes became the gold standard for environmental phylogenetic inference due to its ability to act as a common ground of comparison between vastly different genomes, and its ability to bypass genomes muddled by HGT, but it is not without its own problems. Clustering by any marker gene from amplified shotgun sequences introduces errors and biases from the aforementioned steps. Kunlin demonstrated that sequencing errors can vastly skew the resulting clusters by increasing the rate of false-positive OTUs (3). Sequencing errors introduced new variance in the sequenced reads to an extent that they were clustered as independent units, when they really belong in single groups (3). Kunlin’s experiment of amplifying and clustering against only a single reference genome (E. coli MG1655), and its results founded the notion of implementing stringent read filtering/trimming and a relatively lowered clustering threshold of 97%, as having a high threshold of 100% sequence similarity splits like clusters that can only be differentiated with variations that is attributed to sequencing error (3). 

	With each iteration and revision of phylogenetic workflows, new limitations are found upon existing methodologies, and solutions to overcome. Building upon the advantages of the OTU clustering methods, a next generation clustering algorithm emerges, promising higher resolution and a more biologically meaningful sequence clusters. Callahan’s group proposes that the relatively newly developed amplicon sequence variants (ASVs) method for clustering, performing better than the existing OTU workflow (clustering by similarity threshold) (4). As Callahan outlines, ASVs resolve sequence differences to the nucleotide level, that can resolve differences with nucleotide order in mind (4). Biological filters may then be applied to further specify our definition of microbial species, and cluster sequences accordingly. This effectively reduces bias against simply a 97% threshold being a strictly computational criterion. 

	From Darwin’s days of empiricism to modern day molecular genetics and bioinformatics, our understanding of microbial species as a concept has come a long way. As much as horizontal gene transfer has muddied the waters when it comes to whole-genome-wide comparisons, HGT is an evolutionarily crucial phenomenon in the preservation of genes as well as our widespread biodiversity on a microbial level. Falkowski describes microbes as “Guardians of metabolism” in his paper “The Microbial Engines That Drive Earth’s Biogeochemical Cycles”. HGT has allowed microbes to share genetic information that may help each-other survive selection pressure and become more fit in the environment (5). In turn, the survival of these organisms preserves and pass-on these transferrable genes so that they remain in circulation and will be available for one day when the genes are needed again. The concept of microbial species in the field of microbial ecology has, for the most part, shunned the existence of HGT and mobile genes for ruining the perfect genetic map of evolutionary history. Our best methods for understanding phylogeny involves selecting for strictly vertical inherited marker genes without consideration for the rich evolutionary history that is ignored in HGT. 

A clear and working concept of microbial species is important for characterization of future unexplored microbial domains, and to push our understanding of the limits of the microbial genome. As important as HGT is in determining the community composition of microbes, the speed of technological and methodological progress has always hindered the ability to resolve HGT relative to VGT. Although we still do not collectively have a flawless working concept of microbial species; through many iterations, we have determined a number of things that the concept is not. We learned early on that microbial species cannot be defined through phenotypic empiricism alone. We also learned that it cannot be fully represented as simply a number of similar genomes. More specifically, not even a number of carefully selected similar sequences in a cluster. With all evidence pointing towards HGT as a major counterbalance to selection pressures, it’s about time we learned that the microbial species concept is not complete without consideration for the horizontal gene transfer phenomenon.  

References:

1)	Welch RA, Burland V, Plunkett III G, et al. 2002. Extensive mosaic structure revealed by the complete genome sequence of uropathogenic Escherichia coli. PNAS 99:17020-17024.
2)	Sogin ML, Morrison HG, Huber JA, et al. 2006. Microbial diversity in the deep sea and the underexplored “rare biosphere”. PNAS 103:12115-12120. 
3)	Kulin V, Engelbrektson A, Ochman H, Hugenholtz P. 2010. Wrinkes in the rare biosphere: pyrosequencing can lead to artificial inflation of diversity estimates. Env Micb 12:118-123. 
4)	Callahan BJ, McMurdie PJ, Holmes SP. 2017. Exact sequence variants should replace operational taxonomic units in marker-gene data analysis. ISME J 11:2639-2643.
5)	Falkowski PG, Fenchel T, Delong EF. 2008. The Microbial Engines That Drive Earth’s Biological Cycles. Science 320:1034-1038. 

## Project 1

(See included project_1.pdf with supplementary materials in current directory)

--------------

# Module 4

* Project_2
    + Completion status:
    + Comments:
    
## Project 2

(See included project_2.pdf with supplementary materials in current directory)
